### DISCUSSION TITLE
How to avoid redirects on failed sign in?

### DISCUSSION THREAD

**Discussion by peculiarity** [nextauthjs/next-auth#878] [Help]
**Your question**
How to avoid redirects on failed sign in?

**What are you trying to do**
I have custom sign in page. I would like to display an error message below the sign in form if the user provided wrong username and password. I create a `fetch` request `onSubmit` but the response is a redirect to `/?` which is weird. I have no idea why it redirects to `/?` and how to get the error.

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [ ] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [X] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by balazsorban44** [comment]
I assume you are using `"credentials"` as your provider. Would https://github.com/nextauthjs/next-auth/pull/1219 solve your issue?
